build-logic/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts

```diff
@@ -0,0 +1,174 @@
+        @Suppress("UnstableApiUsage")
+        vendor.set(JvmVendorSpec.ADOPTIUM)
+    }
+}
+
+configurations.all {
+    // In case published versions of a module are also available, always prefer the local one
+    resolutionStrategy.preferProjectModules()
+}
+
+val internal: Configuration = configurations.create("internal") {
+    isCanBeConsumed = false
+    isCanBeResolved = false
+}
+
+dependencies {
+    "internal"(platform("com.hedera.hashgraph:hedera-dependency-versions"))
+}
+
+sourceSets.all {
+    configurations.getByName(annotationProcessorConfigurationName) {
+        extendsFrom(internal)
+    }
+    configurations.getByName(compileClasspathConfigurationName) {
+        extendsFrom(internal)
+    }
+    configurations.getByName(runtimeClasspathConfigurationName) {
+        extendsFrom(internal)
+    }
+}
+
+tasks.withType<AbstractArchiveTask>().configureEach {
+    isPreserveFileTimestamps = false
+    isReproducibleFileOrder = true
+    fileMode = 664
+    dirMode = 775
+}
+
+tasks.jar {
+    exclude("**/classpath.index")
+}
+
+tasks.withType<JavaCompile>().configureEach {
+    options.encoding = "UTF-8"
+}
+
+tasks.withType<Javadoc>().configureEach {
+    options.encoding = "UTF-8"
+    (options as StandardJavadocDocletOptions)
+        .tags("apiNote:a:API Note:", "implSpec:a:Implementation Requirements:", "implNote:a:Implementation Note:")
+}
+
+testing {
+    @Suppress("UnstableApiUsage")
+    suites {
+        // Configure the normal unit test suite to use JUnit Jupiter.
+        named("test", JvmTestSuite::class) {
+            // Enable JUnit as our test engine
+            useJUnitJupiter()
+            targets.all {
+                testTask {
+                    options {
+                        this as JUnitPlatformOptions
+                        excludeTags(
+                            "TIME_CONSUMING",
+                            "AT_SCALE",
+                            "REMOTE_ONLY",
+                            "HAMMER",
+                            "PERFORMANCE",
+                            "PROFILING_ONLY",
+                            "INFREQUENT_EXEC_ONLY"
+                        )
+                    }
+                    // Increase the heap size for the unit tests
+                    maxHeapSize = "4096m"
+                    jvmArgs("-XX:ActiveProcessorCount=16")
+                    // Can be useful to set in some cases
+                    // testLogging.showStandardStreams = true
+                }
+            }
+
+            // Add 'targets' (tasks of type Test) for the 'test' test suite: 'hammerTest' and 'performanceTest'.
+            // Gradle's test suite API does not yet allow to add additional targets there (although it's planned).
+            // Thus, we need to go directly via 'tasks.register' and then set 'testClassesDirs' and 'sources' to
+            // the information we get from the 'source set' (sources) of this 'test suite'.
```

Comment:
This is an improvement in the test setup. Before, there were multiple _source sets_ pointing at the same sources. Which meant the test code was compiled multiple times.

Something was also not completely right with the `includeTags` I believe. If found tests marked with "performance" that are broken (I fixed one, and disabled the others). Seems like these were not (always) executed before.